src/main/java/seedu/address/model/wedding/Date.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.wedding;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+
+import seedu.address.model.person.Name;
+
+/**
+ * Represents the Date of a Wedding.
+ */
+public class Date {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should be in the following format, "
+                    + "YYYY-MM-DD.";
+    public static final String VALIDATION_REGEX = "\\b(\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01]))\\b";
```

Comment:
I think using inbuilt java library DateTimeFormatter may be more convenient, can avoid the regex checks and has it own methods available to use?